app/helpers/manipulation_helper.rb

```diff
@@ -0,0 +1,14 @@
+module ManipulationHelper
+  def manipulate_storage_systems(assoc, storage_systems = nil)
+    if storage_systems.instance_of?(Array) && storage_systems.length >= 1
```

Comment:
What is the storage_systems.instance_of?(Array) check actually checking for?  Are you really just checking for nil?

Reply:
Right, but there are other ways of doing that like checking `respond_to?(:length)`.  But what are you seeing coming in aside from the nil?  or is it only the nil?